src/planner/insert_plan.cpp

```diff
@@ -39,25 +39,29 @@ InsertPlan::InsertPlan(storage::DataTable *table,
       auto &values = (*insert_values)[tuple_idx];
       PL_ASSERT(values.size() <= schema->GetColumnCount());
       uint32_t param_idx = 0;
-      for (uint32_t column_id = 0; column_id < values.size(); column_id++) {
-        auto &exp = values[column_id];
+      for (uint32_t column_id = 0; column_id < schema->GetColumnCount(); column_id++) {
+        auto index = column_id;
```

Comment:
I had to use an `index` variable which becomes `0` for values which are not specified in the `INSERT` (i.e., `id2` for a table `foo(id int, id2 int)` for, say, `INSERT INTO foo VALUES(3)`). Following would've been a more elegant approach:

```
for (uint32_t column_id = 0; column_id < schema->GetColumnCount(); column_id++) {
        if (column_id < values.size()){
          auto &expr = values[column_id];
        }
        const type::TypeId type = schema->GetType(column_id);
        if (column_id >= values.size() || expr == nullptr) {  // expr isn't checked when 1st condition is true
        ...
        ...
```

However, that resulted in a compilation error because of the declaration within an `if` lead to `unidentified error` for `expr` in the code that followed.

Furthermore, I wasn't able to get around this by declaring `expr` outside the `if` and assigning it within the `if` because `expr` requires initialization while declaration.

So, I think this is the only way to prevent the error mentioned and it's working as per my tests. Please suggest a more elegant approach if possible.

Reply:
@pmenon Is there a way to do this thing in `query_to_operator_transformer.cpp`?

Reply:
What is the problem preventing these checks being done in the query_to_operator_transformer.cpp?

Reply:
@pervazea They were simply easy here. I'll try migrating them to `query_to_operator_transformer.cpp` and let you know the results :)